tests/conftest.py

```diff
@@ -0,0 +1,511 @@
+
+    def __iter__(self) -> Iterator[tuple[str, Result]]:
+        return iter(self.results.items())
+
+
+@dataclass
+class TrainedDecodanda:
+    """
+    Encapsulation of a trained decoder for providing to test functions.
+    """
+    data: Data
+    decoder: Decodanda
+    dichotomies: dict[str, list]
+    key: tuple[str, str]  # data_key, decoder_key
+    parameters: dict[str, Any]
+    results: Results
+    trained: bool = False
+
+
+"""
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Registries for provided efficient access to synthetic data, trained decoders, and results
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+"""
+
+
+class DataRegistry:
+    """
+    Persistent registry of synthetic data to be accessed by test functions without
+    the need to regenerate data for each test.
+    """
+    # persistent registry of synthetic data
+    __registry = {}
+
+    @classmethod
+    def get(cls, key: str) -> Data:
+        if key in cls.__registry:
+            if not isinstance(cls.__registry.get(key), Data):
+               cls.__registry[key] = cls.__registry.get(key)()
+            return cls.__registry.get(key)
+        else:
+            raise KeyError(f"Data key '{key}' not found in registry.")
+
+    @classmethod
+    def register(cls, alias: Optional[str] = None):  # noqa: ANN206
+        """
+        Register a synthetic dataset for testing.
+        """
+        def register_data(constructor: Callable):  # noqa: ANN206, ANN001, ANN201
+            nonlocal alias
+            alias = alias if alias else constructor.__name__.split("_")[0]
+            cls.__registry[alias] = constructor
+
+        return register_data
+
+
+class DecodandaRegistry:
+    """
+    Registry of decoder constructors to be accessed by the trained decoder registry (not the test functions).
+    """
+    # persistent registry of decoder constructors
+    __registry = {}
+
+    @classmethod
+    def get(cls, key: str, data: Data) -> TrainedDecodanda:
+        if key in cls.__registry:
+            return cls.__registry.get(key)(data)
+        else:
+            raise KeyError(f"Decoder key '{key}' not found in registry.")
+
+    @classmethod
+    def register(cls, alias: Optional[str] = None):  # noqa: ANN206
+        """
+        Register a synthetic dataset for testing.
+        """
+        def register_data(constructor: Callable):  # noqa: ANN206, ANN001, ANN201
+            nonlocal alias
+            alias = alias if alias else constructor.__name__.split("_")[0]
+            cls.__registry[alias] = constructor
+
+        return register_data
+
+
+class TrainedDecodandaRegistry:
+    """
+    Persistent registry of trained decoders to be accessed by test functions without
+    the need to retrain decoders for each test.
+    """
+    # persistent registry of trained decoders
```

Comment:
update descriptions: these are Decodanda objects containing trained classifiers. In theory "Decodanda" is a class containing the data subdivided into conditions. The fact that it contains the trained classifier is secondary to its role in data handling.

Reply:
In fact, Decodanda does not even have a public "train()" function; it only has a private one that gets called by the various decoding and CCGP routines.

Reply:
**This registry has been removed.** See details in the new pull request (I'm closing this one for reasons stated in new pull request).